challenges/largest_product/largest_product_array.py

```diff
@@ -0,0 +1,33 @@
+def largest_product(arr):
+    """
+    Accepts matrix array and return the largest product of
+    2 adjacent values within the 2D array.
+    """
+    max = arr[0][0] * arr[1][0]
+    row = 0
+    for i in range(len(arr) - 1):
+        col = 0
+        arr = arr[row][0] * arr[row + 1][0]
+        if arr > max:
+            max = arr
+        for _ in range(len(arr[0]) - 1):
+            arr1 = arr[row][col] * arr[row][col + 1]
+            arr2 = arr[row + 1][col] * arr[row + 1][col + 1]
+            arr3 = arr[row][col + 1] * arr[row + 1][col + 1]
+            arr4 = arr[row][col] * arr[row + 1][col + 1]
```

Comment:
This is checking diagonally adjacent values. You only need values that are horizontally and vertically next to each other.